### DISCUSSION TITLE
UniGetUI 3.1.0 beta 0

### DISCUSSION THREAD

**Discussion by marticliment** [Devolutions/UniGetUI#2342] [Announcements]
![](https://img.shields.io/github/downloads/marticliment/wingetui/3.1.0-beta0/total?style=for-the-badge&color=red)

<b>IMPORTANT:
This is the first beta release to implement the UniGetUI rebranding.
Instaling this release will automatically migrate settings and other data to new folders (WingetUI -> UniGetUI), so rolling back to older versions of WingetUI may cause settings to be reset.
</b>

Note: When it comes to features, this release should be quite similar to the stable 3.1.0 release
## Changes 
 - Packages will be loaded and stored by a Package Loader, instead of being loaded by the interface. 
 - Package lists have been rebuilt to use ItemsView, causing UniGetUI to use less ram (~290mb max -> ~210mb max)
 - The Widgets API has been optimized. Packages load up to way faster on the widgets.
 - The Package Manager logs have been rebuilt from the grounds up.
   - Package Manager logs for each manager can be isolated
   - Normal and Verbose views available
 - Improvements to keyboard navigation
 - Chocolatey can be set to not be installed by the UniGetUI Installer
 - Lots of internal code changes and optimizations.
 - Other bugfixes

SHA256: `79D8266736B46D8B8959DD775702B275AE4B1936EE6485D01C31A1F62418B8DB`

<hr /><em>This discussion was created from the release <a href='https://github.com/marticliment/WingetUI/releases/tag/3.1.0-beta0'>UniGetUI 3.1.0 beta 0</a>.</em>

**Comment by MarcLaf** [comment]
Just tried this release and none of the Winget updates are showing, only PowerShell Gallery. Tried disabling and re-enabling Winget but no good.

**Reply by marticliment** [reply]
can you please share your logs and your winget package manager logs?

**Reply by MarcLaf** [reply]
Sure thing.
[PackageManager Log.txt](https://github.com/user-attachments/files/15991326/PackageManager.Log.txt)
[UniGetUI Log.txt](https://github.com/user-attachments/files/15991328/UniGetUI.Log.txt)

**Reply by marticliment** [reply]
please run on a powershell window
```
Install-Module Microsoft.WinGet.Client
```
and try to run UniGetUI again

**Reply by nwesterhausen** [reply]
Having a similar issue, so I'll just add as a comment here. After install, detected choco packages fine (probably because it imports them?) but failed to detect scoop and winget. 

I had UniGet install scoop. That seemed to work for scoop.

I saw the Install-Module command suggested here for the winget problem, so did that, both in normal and elevated window (after it didn't make a difference in the normal powershell). No dice. Still doesn't detect WinGet stuff and logs an error about it missing that specific module even tho I installed it.

[UniGetUI PackageManager Log.txt](https://github.com/user-attachments/files/16012636/UniGetUI.PackageManager.Log.txt)
[UniGetUI Log.txt](https://github.com/user-attachments/files/16012637/UniGetUI.Log.txt)

**Reply by MarcLaf** [reply]
> please run on a powershell window
> 
> ```
> Install-Module Microsoft.WinGet.Client
> ```
> 
> and try to run UniGetUI again

I also did it in a non-admin and admin ps window, still no results for WinGet packages.
[PackageManager Log - WingetVerbose.txt](https://github.com/user-attachments/files/16015594/PackageManager.Log.-.WingetVerbose.txt)
[UniGetUI Log2.txt](https://github.com/user-attachments/files/16015595/UniGetUI.Log2.txt)

**Reply by RobertoPuraVida** [reply]
Hi Marc. I uninstalled WinGetUI/UniGETUI. Then in ADMIN Powersell, I did the Winget.Client reinstallation. Than reinstallet UniGETUI. 
It took a few Minutes after the reinstallation, but suddenly the installed packages showwed up. Hope it will work for you.

**Reply by nwesterhausen** [reply]
So maybe what the problem is that I use Powershell 7 by default? I told Terminal to open up the older version of powershell and I can replicate the `Get-WinGetPackage` command failure. So I opened that up as admin, ran the install for Microsoft.WinGet.Client, uninstalled UniGetUI, reinstalled UniGetUI and now it seems to be pulling in WinGet packages correctly.

With the previous version (I think I was on WinGetUI 3.0.2?) no issues, just with this Beta0.

So yes, now everything is working, but only after opening the older version of powershell as admin, installing the client, and reinstalling.

![image](https://github.com/marticliment/WingetUI/assets/2317381/759ccf27-3190-4193-b3fa-154c0173a125)

**Reply by marticliment** [reply]
if you run powershell.exe which version does start?
Running `powershell.exe -Version 5` does launch Windows PowerShell (PowerShell v5)?

**Reply by MarcLaf** [reply]
Ohh I use PS7 by default for everything and only Windows PowerShell when absolutely necessary. I'll try it again.

**Reply by MarcLaf** [reply]
Still not working for me - Might just have to remove and roll back to WinGetUI

**Reply by MarcLaf** [reply]
Rolled back to WinGetUI and Winget packages work but Powershell modules update is now broken :-(
Ah well, I'll get it straightened out eventually!

**Reply by leowankerddd** [reply]
@marticliment ,

I also was missing every winget program after upgrading/installing the beta.

I ran the command you recommended and then re-ran the beta and everything was there. Thanks.
Install-Module Microsoft.WinGet.Client

Windows told me that I was installing from an untrusted Repository.
![ksnip_20240628-231823](https://github.com/marticliment/WingetUI/assets/61802928/b2a75c88-8d6a-434b-8938-2fe72987e88e)

**Comment by RobertoPuraVida** [comment]
Testing the Beta0. Still, no “installed Applications” are visible. No SOURCES are shown. Like with alpha0 and alpha1. Back to 3.0.2 and all is okay again. Tested as well the Beta0 as portable installation, same result: No installed applications found.

**Reply by marticliment** [reply]
please run on a powershell window
```
Install-Module Microsoft.WinGet.Client
```
and try to run UniGetUI again

**Reply by RobertoPuraVida** [reply]
Yes! The reinstallation of the WinGet Client solved the problem for me. As I tried  UniGetUI Beta0 on two computers and had the same issue (no installed packages), this should be somehow be fixed in the final version of UniGetUI. 

Thanks a lot for all your work!!!!

**Reply by marticliment** [reply]
Are you using PowerShell7 on your system?

**Reply by RobertoPuraVida** [reply]
PSVersion                      5.1.22621.3646

Okay, maybe this was the issue?

**Reply by marticliment** [reply]
No, if using PowerShell 5 it should autoinstall.

Please try the following:
1. Uninstall the `Microsoft.WinGet.Client` module: `Uninstall-Module Microsoft.WinGet.Client`. UniGetUI should stop showing packages
2. Run the contents of this file into a powershell window, and attach here the output: https://github.com/marticliment/WingetUI/blob/main/InstallerExtras/EnsureWinGet.ps1

**Reply by RobertoPuraVida** [reply]
.\EnsureWinGet.ps1 : Die Datei "C:\Users\MeUser\EnsureWinGet.ps1" kann nicht geladen werden. Die Datei
"C:\Users\MeUser\EnsureWinGet.ps1" ist nicht digital signiert. Sie können dieses Skript im aktuellen System nicht
ausführen. Weitere Informationen zum Ausführen von Skripts und Festlegen der Ausführungsrichtlinie erhalten Sie unter
"about_Execution_Policies" (https:/go.microsoft.com/fwlink/?LinkID=135170)..
In Zeile:1 Zeichen:1
+ .\EnsureWinGet.ps1
+ ~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : Sicherheitsfehler: (:) [], PSSecurityException
    + FullyQualifiedErrorId : UnauthorizedAccess

**Reply by marticliment** [reply]
And running `powershell.exe -ExecutionPolicy Bypass -File ".\EnsureWinGet.ps1"`?

**Reply by marticliment** [reply]
And if you paste the contents of the EnsureWinGet.ps1 file directly into the powershell, what happens?

**Reply by RobertoPuraVida** [reply]
C:\Users\Robert>powershell -ExecutionPolicy Bypass -File ".\EnsureWinGet.ps1"
Microsoft.WinGet.Client is not installed or is not on the latest version
Attempting to uninstall an older version of Microsoft.WinGet.Client...
PackageManagement\Uninstall-Package : Es wurde keine Übereinstimmung für die angegebenen Suchkriterien und Modulnamen
'Microsoft.WinGet.Client' gefunden.
In C:\Program Files\WindowsPowerShell\Modules\PowerShellGet\1.0.0.1\PSModule.psm1:2194 Zeichen:21
+ ...        $null = PackageManagement\Uninstall-Package @PSBoundParameters
+                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Microsoft.Power...ninstallPackage:UninstallPackage) [Uninstall-Package]
   , Exception
    + FullyQualifiedErrorId : NoMatchFound,Microsoft.PowerShell.PackageManagement.Cmdlets.UninstallPackage

Installing Microsoft.WinGet.Client...

Name                           Version          Source           Summary
----                           -------          ------           -------
nuget                          2.8.5.208        https://onege... NuGet provider for the OneGet meta-package manager
Microsoft.WinGet.Client was installed successfully
Checking for updates for Microsoft.WinGet.Client module...
Microsoft.WinGet.Client is up-to-date
Checking for updates for Microsoft.WinGet.Client module...
Microsoft.WinGet.Client is up-to-date

**Reply by RobertoPuraVida** [reply]
I first uninstalled the Modul winget.client and let the script run. UniGetUI is still delivering the installed packages.

**Reply by marticliment** [reply]
Whey you say you use PowerShell 7 by default, `powershell.exe` does start powershell 7 or Windows PoserShell 5?

**Comment by BroMarduk** [comment]
*Edited to reflect proper PowerShell versions

Having similar issues.  I have 4 versions of PowerShell 5 x64, 5 x86, 7 x64 and 7 x86.  Installed Microsoft.WinGet.Client in each.
Uninstalled WinGetUI, installed UniGetUI...

![image](https://github.com/marticliment/WingetUI/assets/2371774/1d121656-7324-4e38-a779-71c03249484d)

If I run the EnsureWInGet PS script, I get 

`Microsoft.WinGet.Client is up-to-date` 

a couple of times in each PowerShell window.

Any suggestions?

**Reply by marticliment** [reply]
I need more info about the Multiple PowerShell Versions:
 - the command `powershell.exe` which shell launches?
 - On which PWSH version are you running the EnsureWinGetClient script?
 - The PowerShell gallery manager on UniGetUI, which version reports?

**Reply by BroMarduk** [reply]
Powershell 7 x64 runs by default if I type from Start Menu, If I type powershell.exe in a Command Prompt it is 5.1 x64

I ran EnsureWinGetClient script on all 4 (PS7 x64, PS7 x86, PS5 x64, PS5 x86).  I got the up-to-date message.

This is what I get for PowerShell Versions:

![image](https://github.com/marticliment/WingetUI/assets/2371774/9ca85a43-667f-44eb-b9b0-686548b10fb5)

**Comment by skanda890** [comment]
None of the packages from Winget and Local are being shown

**Reply by marticliment** [reply]
Can you please attach WinGet Logs?

**Reply by skanda890** [reply]
> Can you please attach WinGet Logs?

See #2344 and #2007

**Reply by marticliment** [reply]
Try to run the command `Install-Module Microsoft.WinGet.Client` on a powershell window

**Comment by BroMarduk** [comment]
Latest Beta 3.1.0 Beta 1 fixed my issues.
